src/entrypoint.S

```diff
@@ -9,9 +9,9 @@
 	.section .boot
 	.global _start
 _start:
-	lw t0, 0x80000318			/* memory size */
-	li t1, 0x7FFFFFF0
-	addu sp,t0,t1				/* init stack */
+	jal get_memory_size
```

Comment:
We can't call C code from here, we have not yet initialized SP and GP, and both of them are necessary to be able to call arbitrary C code.

Also, I'm not sure what it is the benefit. The idea of `get_memory_size` is that someone would call `sys_set_boot_cic` before calling it, and surely nobody can call that before `_start`. 

I don't like very much the `sys_set_boot_cic` thing, specifically because it doesn't solve stack placement (which is #76). I think it's been a mistake to introduce a runtime API. It should have been a compile-time option. Or if we want to do a runtime thing, it must be an autodetection in `entrypoint.S`. For instance, the code could checksum the IPL3 boot code (0x1000_0040 - 0x1000_1000) [like emulators do](https://github.com/rasky/r64emu/blob/4902c2b942ad1be36fd7bb662c93381fe10d5566/src/cartridge.rs#L64-L72) to try to guess the correct CIC model.

So, unless you want to start a new adventure, I'd remove this commit.

Reply:
Haha you are right, I bodged this in the last minute 😅 sp is easy to initialize once more before calling it as we know the minimum memory amount and we can also prepare gp but I didn't know `get_memory_size` depended on user-set info either so this does not make sense. I will remove this nonsense 😊
Even the invalidate all functions are rather useless 🤷🏻‍♂️